src/web/components/elements/WitnessVideoBlockComponent.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { css } from 'emotion';
+
+import { space } from '@guardian/src-foundations';
+import { headline, body } from '@guardian/src-foundations/typography';
+
+import { pillarPalette } from '@root/src/lib/pillars';
+import { WitnessWrapper } from '@frontend/web/components/WitnessWrapper';
+
+type Props = {
+	title: string;
+	description: string;
+	authorUsername: string;
+	width: number;
+	height: number;
+	youtubeUrl: string;
+	dateCreated: string;
+	pillar: Theme;
+};
+
+const captionStyles = css`
+	margin-top: ${space[3]}px;
+`;
+
+const titleStyles = (pillar: Theme) => css`
+	margin-bottom: ${space[2]}px;
+	color: ${pillarPalette[pillar].main};
+	${headline.xxxsmall()}
+`;
+
+export const WitnessVideoBlockComponent = ({
+	title,
+	description,
+	authorUsername,
+	width,
+	height,
+	youtubeUrl,
+	dateCreated,
+	pillar,
+}: Props) => {
+	const youtubeVideoId = youtubeUrl.split('?v=')[1];
```

Comment:
This seems like it might cause issues. Is there a better way to do this?

Reply:
The only placed where this is defined would be the `html` or `youtubeUrl` props. The `html` option isnt ideal for styling. The `youtubeUrl` has some issues, the example in storybooks (extracted from CAPI) retrurns a string like: `'http://www.youtube.com/watch?v=XRq83txBgxw'`. It appears when you try load a video in an iframe using `/watch` that it will not load, instead you need to use `/embed/${videoId}` to get it to work.

The only way I see of extracting the youtuve video ID is to extract it via `youtubeUrl`.